libs/cli/src/main/java/org/elasticsearch/cli/internal/JsonPrintWriter.java

```diff
@@ -0,0 +1,142 @@
+
+    private final StringBuilder buffer = new StringBuilder(INITIAL_BUFFER_SIZE);
+    private final Clock clock;
+
+    public JsonPrintWriter(OutputStream out, boolean autoFlush) {
+        this(out, autoFlush, Clock.systemUTC());
+    }
+
+    @SuppressForbidden(reason = "Override PrintWriter to emit Json")
+    protected JsonPrintWriter(OutputStream out, boolean autoFlush, Clock clock) {
+        super(out, autoFlush);
+        this.clock = clock;
+    }
+
+    @Override
+    public void write(String msg) {
+        if (isJsonObject(msg)) {
+            super.write(msg);
+        } else {
+            synchronized (buffer) {
+                buffer.append(msg);
+            }
+        }
+    }
+
+    @Override
+    public void println(String msg) {
+        final String json;
+        if (isJsonObject(msg)) {
+            // Forward as is, ignore current buffer.
+            json = msg;
+        } else {
+            // Don't mind if another thread might miss a none-empty buffer.
+            // It's highly unlikely this is meant to be concatenated into the same line.
+            json = toJson(buffer.isEmpty() ? msg : getAndClearBuffer(msg), clock.millis());
+        }
+        super.write(json);
+        super.println();
+    }
+
+    @Override
+    public void println() {
+        flush();
+        super.println();
+    }
+
+    @Override
+    public void flush() {
+        synchronized (buffer) {
+            if (buffer.isEmpty() == false) {
+                super.write(toJson(getAndClearBuffer(), clock.millis()));
+            }
+        }
+        super.flush();
+    }
+
+    @Override
+    public void close() {
+        flush();
+        super.close();
+    }
+
+    /**
+     * Converts {@code msg} to JSON using the {@link EcsJsonSerializer}.
+     */
+    protected static String toJson(String msg, long timeMillis) {
+        StringBuilder builder = new StringBuilder(INITIAL_BUFFER_SIZE);
+        EcsJsonSerializer.serializeObjectStart(builder, timeMillis);
+        EcsJsonSerializer.serializeFormattedMessage(builder, msg);
+        stripTrailing(builder, ' ');
+        stripTrailing(builder, ',');
+        EcsJsonSerializer.serializeObjectEnd(builder);
+        stripTrailing(builder, '\n');
+        return builder.toString();
+    }
+
+    private static void stripTrailing(StringBuilder builder, char c) {
+        if (builder.length() > 0 && builder.charAt(builder.length() - 1) == c) {
+            builder.setLength(builder.length() - 1);
+        }
+    }
+
+    /**
+     * Checks if {@code str} is likely a JSON object surrounded by `{` and `}` without any further validation.
+     * Leading and trailing whitespaces are ignored.
+     */
+    protected static boolean isJsonObject(String str) {
+        if (str == null) return false;
+
+        int i = 0;
+        while (i < str.length() && Character.isWhitespace(str.charAt(i))) {
+            i++;
+        }
+        if (i == str.length() || str.charAt(i) != '{') return false;
```

Comment:
Since we control the log format, I think we can assume any line beginning with `{` is json, so we should just be able to return whether the first character is `{` and not need to check the entire string. I'm not even sure we need to check for whitespace, again, since we completely control the format coming from ES.

Reply:
👍 If've simplified this as suggested, also removing whitespace checks